docs/root/configuration/other_features/reverse_tunnel.rst

```diff
@@ -223,228 +169,143 @@ For secure reverse tunnel establishment, add a TLS context to the upstream clust
-reverse connections instead of creating new forward connections.
+The reverse connection cluster is a special cluster type that routes traffic through established reverse
+tunnels rather than creating new outbound connections. When a data request arrives at the upstream Envoy
+for a downstream node, the cluster looks up a cached reverse tunnel connection to that node and reuses it.
 
-For each such data request, a "host_id" is specified or computed from the request headers; this "host_id"
-represents the downstream node that the request should be routed to.
+Each data request must include a ``host_id`` that identifies the target downstream node. This ID can be
+specified directly in request headers or computed from them. The cluster extracts the ``host_id`` using
+the configured ``host_id_format`` field and uses it to look up the appropriate reverse tunnel connection.
 
-.. validated-code-block:: yaml
-  :type-name: envoy.config.cluster.v3.Cluster
+.. literalinclude:: /_configs/reverse_connection/responder-envoy.yaml
+    :language: yaml
+    :lines: 92-112
+    :linenos:
+    :lineno-start: 92
 
-  name: reverse_connection_cluster
-  connect_timeout: 200s
-  lb_policy: CLUSTER_PROVIDED
-  cluster_type:
-    name: envoy.clusters.reverse_connection
-    typed_config:
-      "@type": type.googleapis.com/envoy.extensions.clusters.reverse_connection.v3.ReverseConnectionClusterConfig
-      cleanup_interval: 60s
-      host_id_format: "%REQ(x-computed-host-id)%"
-  typed_extension_protocol_options:
-    envoy.extensions.upstreams.http.v3.HttpProtocolOptions:
-      "@type": type.googleapis.com/envoy.extensions.upstreams.http.v3.HttpProtocolOptions
-      explicit_http_config:
-        http2_protocol_options: {}  # HTTP/2 required for reverse connections
-
-The reverse connection cluster configuration specifies:
-
-* **Load balancing policy**: ``CLUSTER_PROVIDED`` allows the custom cluster to manage load balancing
-* **Host ID Format**: Uses Envoy's formatter system to extract the target downstream node identifier from
-  request context. The ``host_id_format`` field supports:
-
-  - ``%REQ(header-name)%``: Extract value from request header
-  - ``%DYNAMIC_METADATA(namespace:key)%``: Extract value from dynamic metadata
-  - ``%FILTER_STATE(key)%``: Extract value from filter state
-  - ``%DOWNSTREAM_REMOTE_ADDRESS%``: Use downstream connection address
-  - Plain text and combinations of the above
-
-An example of how to process headers and set the host ID is described in the
-:ref:`config_reverse_connection_egress_listener` section.
-
-* **Protocol**: Only HTTP/2 is supported for reverse connections.
-* **Host Reuse**: Once a host is created for a specific downstream node ID, it is cached and reused for all
-  subsequent requests to that node. Therefore, each data request is multiplexed as a new stream on the existing
-  HTTP/2 connection.
+The reverse connection cluster configuration includes several key fields:
+
+**Load balancing policy**
```

Comment:
i think semantically these are headings so probs should use some level of heading